### ISSUE TITLE
mpirun noticed that process rank 1 with PID 0

### ISSUE THREAD

**Issue by mohapatras** [horovod/horovod#2300]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) Tensorflow
2. Framework version: 1.15.0
3. Horovod version: 0.20.0
4. MPI version:3.0
5. CUDA version:10.1
6. NCCL version:2.5
7. Python version:3.7.8
8. OS and version:Ubuntu 18.04
9. GCC version: 7.5.0

**Checklist:**
1. Did you search issues to find if somebody asked this question before? Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
mpi processes are killed. Below is a screenshot:
![mpi](https://user-images.githubusercontent.com/10518386/93736299-83843a00-fbfd-11ea-863d-764d50e1f927.PNG)

Does this happen with all the horovod versions ? CPU is Intel Xeon. I am trying to run with 8 GPUs. This error pops up even with reducing the batch size. Batch sizes are multiples of GPU. Also any documentation on resource usage will be helpful.

**Comment by tgaddair** [comment]
Hey @mohapatras, that MPI message is just saying that one of the worker processes failed, so the whole job terminated.  Specifically, something sent a `kill -9` to one of the worker processes.

Most likely, this is being caused by an out of memory error.  How much memory do you have on your 8 GPU machine?  How big is your dataset?  Are you reading it all into memory or streaming it in in batches?

**Comment by mohapatras** [comment]
@tgaddair Sorry for the late reply. The 8 GPU machine is Nvidia 2070 RTX 11 GB per piece. System RAM is 128 GB. I am integrating horovod make use of this [yolo v3 tensorflow](https://github.com/YunYang1994/tensorflow-yolov3) to work. I am passing a batch size of 16. My input image size is 608X608.

**Comment by mohapatras** [comment]
[hvd_train_py.zip](https://github.com/horovod/horovod/files/5274578/hvd_train_py.zip)

Here is the modified train.py for the zip. Increasing the batch size to 32 or more system runs out of memory. I am trying to achieve a benchmarking on the results with distributed training.

**Comment by tgaddair** [comment]
Do you know how much of the dataset each worker is reading into memory?  As an extreme example, if every worker reads in the full dataset, then using Horovod with 8GPUs will require 8x the system RAM.  Most systems will be stream in the data, but this could be one source of the OOM error.

**Comment by mohapatras** [comment]
I couldn't find quantity of dataset each worker is reading into memory. But I did played with different GPUs and batch sizes. I can successfully run using 2 GPUs with total batch size of 12(usually batch size 6 for single GPU, more than that throws out of memory error). However the speed is too slow while covering the training steps in an epoch. System RAM consumption is also too less. 16/128 GB. Also the training step is full length of training set.

Do I need to change anything on the dataset reading ?
Are the hooks correct in the train.py in above zip I shared ?

Just thinking if I am missing any configuration values.
